Muehle_Logic.ipynb

```diff
@@ -237,6 +237,56 @@
     "            positions.append((ring + 1, cell))\n",
     "    return positions"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_mills():\n",
+    "    global status\n",
+    "    pieces = status.pieces\n",
+    "    player = status.current_player\n",
+    "    oldMills = status.mills\n",
+    "    newMills = set()\n",
+    "    match = set()\n",
+    "    currentPlayerMills = set()\n",
+    "\n",
+    "    #find all current mills on the board\n",
+    "    for x in range (0,3):\n",
+    "        for i in [0,2,4,6]:\n",
+    "                if(pieces[1][x][i] == pieces[1][x][i+1] == pieces[1][x][(i+2) % 8] != 0):\n",
+    "                    newMills.add((pieces[1][x][i], x ,i))\n",
+    "\n",
+    "    for i in [1,3,5,7]:\n",
+    "        if(pieces[1][0][i] == pieces[1][2][i] == pieces[1][2][i] != 0):\n",
+    "            newMills.add((pieces[1][x][i], 0 ,i))\n",
+    "\n",
+    "    #set new mills\n",
+    "    status.mills = newMills   \n",
+    "            \n",
+    "\n",
+    "    #if the current mills are the same mills as the old ones or there are now less mills on the board => no new mills\n",
+    "    if(oldMills == newMills or len(oldMills) > len(newMills)):\n",
+    "        return(0)\n",
+    "\n",
+    "\n",
+    "    #find new mills for current player\n",
+    "    if(oldMills != newMills):\n",
+    "        for mill in newMills:\n",
+    "            for oldMill in oldMills:\n",
+    "                if mill == oldMill:\n",
+    "                    match.add(mill) \n",
+    "    newMills = newMills.difference(match)\n",
```

Comment:
ist es Absicht, dass du newMills wieder verwendest und die Differenz nicht in einer neuen Variable speicherst? Tendenziell ist ja beides möglich. Mir egal.

Reply:
zuerst stehen in new_mills alle gefundenen Mühlen. Erst durch difference mit den schon den vorher existierenden erhalte ich die Neuen.